### ISSUE TITLE
Clearer diagnostic when a target server drops mid-deploy

### ISSUE THREAD

**Issue by drzacharysmith68** [Schema-Smith/SchemaSmithyFree#355]
### Summary

When a target database server drops **mid-deploy** — it restarts, crashes, or is OOM-killed *after* SchemaQuench has already connected and begun work — the failure currently surfaces as a raw, misleading provider error. For SQL Server it looks like:

```
Cannot continue the execution because the session is in the kill state.
 ---> Microsoft.Data.SqlClient.SqlException ... (provider: TCP Provider, error: 35 - An internal exception was caught)
 ---> System.Net.Sockets.SocketException (0x80004005): Success
    at ... [full ADO.NET stack]
```

`SocketException: Success` is nonsensical to a user, and the stack gives no hint that the *server went away* rather than a script being broken or credentials being wrong.

### Proposed change

Detect a mid-run connection loss (distinct from an initial connect failure, which the pre-flight connection test already reports clearly) and surface a purpose-built message that names the server and phase and points at the real cause — across SQL Server, PostgreSQL, and MySQL. For example:

> Lost connection to `demoserver` during `Template:Northwind` — the server stopped responding mid-deploy (it may have restarted, crashed, or run out of memory). This is an environment problem, not a schema error: check the server/container logs and available memory, then re-run — SchemaQuench is idempotent and will converge cleanly.

The full provider stack is preserved in the error log; only the lead message changes.

### Context

Motivated by the diagnosis in #353, where a memory-constrained Docker backend OOM-killed SQL Server mid-kindling and the raw `SocketException: Success` obscured the real (environment) cause.

**Comment by drzacharysmith68** [comment]
Fixed in #357 (merged to main 2026-07-20, merge commit 4bae0764). Mid-deploy connection loss is now classified by a tri-engine ConnectionLostClassifier and surfaced as a purpose-built "lost connection to <server> during <phase> — environment problem, re-run" message across SQL Server, PostgreSQL, and MySQL; the raw provider stack is preserved in the error log.

(Auto-close didn't fire because the closing keyword was inside a code span in the PR body — closing manually.)
